chafa/chafa-term-db.c

```diff
@@ -267,6 +267,17 @@ static const SeqStr *color_16_list [] =
     NULL
 };
 
+static const SeqStr *color_8_list [] =
+{
+    color_8_seqs,
+    NULL
+};
```

Comment:
@hpjansson I tried to follow the pattern of `color_16_list` in order to add support for only 8 colors, but this didn’t work as expected.

When I use `TERM=eat-color`, which uses `color_8_list`, the result is actually monochrome:
![TERM=eat-color](https://github.com/hpjansson/chafa/assets/155692/350da6d5-553c-44d3-b9e6-c5890d56a991)

But when I ask explicitly for 8 colors, I get what I expect:
![chafa -c](https://github.com/hpjansson/chafa/assets/155692/197380cb-d055-4c5d-b05d-33272c8d8d87)

I’m obviously missing something, but I am at a loss for how `ChafaTermInfo` influences the color palette.

Reply:
The code you've added to `ChafaTermInfo` looks correct. The missing piece goes in the CLI tool, between `CHAFA_CANVAS_MODE_INDEXED_16` and `CHAFA_CANVAS_MODE_FGBG_BGFG`:

https://github.com/hpjansson/chafa/blob/c12ef2ca180b43988ed1755e4d719cc3e57a29ef/tools/chafa/chafa.c#L1752-L1768

It needs a case for `CHAFA_CANVAS_MODE_INDEXED_8`.

In the long term, I'd like to add library API to populate a `ChafaCanvasConfig` from a `ChafaTermInfo`, but I've been putting it off because it requires some careful thinking (a bad API would be worse than none).

Reply:
That was exactly right! Thanks for your help. This has been fixed by [`6824130..2126cd9`](https://github.com/hpjansson/chafa/compare/6824130075416ed97edbd6020677152be8517b9f..2126cd9bd7289791d0d49e187e9e4a16ed62d3e5).

Reply:
Great!